lang/llvm-12/Portfile

```diff
@@ -0,0 +1,306 @@
+                        your code, and is a great example of the sort of tool \
+                        that can be built using the Clang frontend as a \
+                        library to parse C/C++ code.
+
+    depends_lib-append  port:libedit port:libffi port:ncurses path:lib/libxar.dylib:xar port:zlib port:llvm-${llvm_version}
+    depends_run-append  bin:perl:perl5 port:clang_select
+
+    # not yet tested and patches need to be moved
+    if {${os.platform} eq "darwin" && ${os.major} < 17} {
+        depends_build
+        depends_lib
+        depends_run
+        archive_sites
+        known_fail yes
+        pre-fetch {
+            ui_error "${subport} is not supported on this os version at present."
+            return -code error {unsupported platform}
+        }
+    }
+
+    # CMAKE_LINKER is used to determine the value for HOST_LINK_VERSION
+    configure.args-append \
+        -DCMAKE_LINKER=${prefix}/bin/ld    \
+        -DCLANG_INCLUDE_TESTS=OFF          \
+        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+        -DCLANG_ENABLE_ARCMT=OFF           \
+        -DDARWIN_PREFER_PUBLIC_SDK=ON      \
+        -DLLVM_BUILD_RUNTIME=ON            \
+        -DLIBCXX_ENABLE_SHARED=ON          \
+        -DLIBCXX_INSTALL_LIBRARY=ON
+
+    # clang project requires libcxx project
+    # else the built 1st stage compiler tries to link against libstdc++ and finds an old version
+    # flang project requires mlir project
+    configure.args-append   \
+        -DLLVM_ENABLE_PROJECTS="clang\;clang-tools-extra\;compiler-rt\;libcxx\;libcxxabi\;flang\;mlir\;lld\;openmp\;libunwind"
+}
+
+if {${subport} eq "lldb-${llvm_version}"} {
+
+    homepage            https://lldb.llvm.org/
+    description         the LLVM debugger
+    long_description    Lldb is the "LLVM native" debugger.
+
+    depends_lib-append  port:libedit port:libffi port:ncurses path:lib/libxar.dylib:xar port:zlib
+    depends_lib-append  port:llvm-${llvm_version} port:clang-${llvm_version}
+    depends_run-append  bin:perl:perl5
+
+    if {${os.platform} eq "darwin" && ${os.major} < 17} {
+        depends_build
+        depends_lib
+        depends_run
+        archive_sites
+        known_fail yes
+        pre-fetch {
+            ui_error "${subport} is not supported on this os version at present."
+            return -code error {unsupported platform}
+        }
+    }
+
+    set worksrcpath ${workpath}/${distname}/lldb
+
+    configure.args-append \
+        -DLLDB_CODESIGN_IDENTITY=- \
+        -DLLDB_ENABLE_LUA=OFF
+
+}
+
+variant tests description {Enable building of test code} {
+    depends_test-append     port:py39-psutil
+    test.run                yes
+    configure.args-append   -DLLVM_INCLUDE_TESTS=ON
+    test.target             check
+
+# this should work for lldb, but doesn't as gmock is not being found
+#   if {${subport} eq "lldb-${llvm_version}"} {
+#       test.target    check-lldb
+#   }
```

Comment:
I'm not 100% sure how I'm going to fix this lldb testing issue. We use llvmtree/lldb as the source root only for the lldb build because lldb has no good unique install targets and otherwise we would have to build all of llvm/clang/compiler-rt to build lldb. I guess perhaps that is just what I will do, even though it's a bit time consuming to do it that way. Probably < 5 users even use the lldb port on MacPorts, I would imagine...I'd be curious to know --- hah! -- it's only 4 (probably all me :> ) <https://ports.macports.org/port/lldb-11/stats?days=30&days_ago=0>